api/src/api/article/content-types/article/lifecycles.js

```diff
@@ -0,0 +1,214 @@
+const addOrCreateStage = (entity, newStage) => {
+  if (entity.publicationStages != null) {
+    return {
+      stages: entity.publicationStages.stages.concat(newStage),
+    };
+  } else {
+    return { stages: [newStage] };
+  }
+};
+
+module.exports = {
+  async beforeUpdate(event) {
+    const { params } = event;
+
+    const entity = await strapi.db.query('api::article.article').findOne({
```

Comment:
should use entityService here, no reason to use the query engine. Best practice guidelines that we should dog food and show users the "recommended" way to do things